### PR TITLE
Upgrade rails 7.2

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made?

Rails 7.2 upgrade

## How was this change tested?

Spec and QA

~~HOLD for more testing on UAT/QA~~

### CODE REVIEW

File: app/models/publication.rb [hunk 1]
```diff
@@ -103,8 +103,7 @@ def self.build_new_manual_publication(pub_hash, original_source_string)
   # @return [self]
   def update_manual_pub_from_pub_hash(incoming_pub_hash, original_source_string)
     self.pub_hash = incoming_pub_hash.merge(provenance: Settings.cap_provenance)
-    match = UserSubmittedSourceRecord.find_by_source_data(original_source_string)
-    match.publication = self if match # we may still throw this out w/o saving
+    match = UserSubmittedSourceRecord.find_by(source_data: original_source_string)
```
Comment (peetucket) [inline app/models/publication.rb:106]: more idiomatic finder

File: app/models/publication.rb [hunk 2]
```diff
@@ -103,8 +103,7 @@ def self.build_new_manual_publication(pub_hash, original_source_string)
   # @return [self]
   def update_manual_pub_from_pub_hash(incoming_pub_hash, original_source_string)
     self.pub_hash = incoming_pub_hash.merge(provenance: Settings.cap_provenance)
-    match = UserSubmittedSourceRecord.find_by_source_data(original_source_string)
-    match.publication = self if match # we may still throw this out w/o saving
```
Comment (peetucket) [inline app/models/publication.rb:107]: line 107 is not actually needed and causes a problem in Rails 7.2

we are setting the relationship below in line 114: https://github.com/sul-dlss/sul_pub/pull/1741/files#diff-25593f507bc85d4e5693ced507f17f74c2ab82adacb44eb124154e1dddac7c73R114

We specify an "inverse_of" relationship which takes care of this automatically: https://github.com/sul-dlss/sul_pub/blob/main/app/models/user_submitted_source_record.rb#L5

we

File: bin/setup [hunk 3]
```diff
@@ -1,11 +1,11 @@
 #!/usr/bin/env ruby
 require "fileutils"
 
-# path to your application root.
 APP_ROOT = File.expand_path("..", __dir__)
+APP_NAME = "sulbib"
```
Comment (justinlittman) [inline bin/setup:5]: sulbib?

File: config/application.rb [hunk 4]
```diff
@@ -9,19 +9,19 @@
...
+    # Please, add to the `ignore` list any other `lib` subdirectories that do
+    # not contain `.rb` files, or that should not be reloaded or eager loaded.
+    # Common ones are `templates`, `generators`, or `middleware`, for example.
+    config.autoload_lib(ignore: %w[assets tasks])
 
-    config.active_job.queue_adapter = :async
-
-    # https://discuss.rubyonrails.org/t/cve-2022-32224-possible-rce-escalation-bug-with-serialized-columns-in-active-record/81017
-    # Used in publications.pub_hash
-    config.active_record.yaml_column_permitted_classes = [Symbol, Hashie::Mash, Hashie::Array]
```
Comment (justinlittman) [inline config/application.rb:25]: Sure you don't need this anymore?

File: config/environments/production.rb [hunk 5]
```diff
@@ -13,16 +13,15 @@
...
 
-  # Ensures that a master key has been made available in either ENV["RAILS_MASTER_KEY"]
-  # or in config/master.key. This key is used to decrypt credentials (and other encrypted files).
+  # Ensures that a master key has been made available in ENV["RAILS_MASTER_KEY"], config/master.key, or an environment
+  # key such as config/credentials/production.key. This key is used to decrypt credentials (and other encrypted files).
   # config.require_master_key = true
 
-  # Disable serving static files from the `/public` folder by default since
-  # Apache or NGINX already handles this.
-  config.public_file_server.enabled = ENV["RAILS_SERVE_STATIC_FILES"].present?
```
Comment (justinlittman) [inline config/environments/production.rb:25]: Should this be carried over? (I've been in the repos I migrated though not sure if necessary.)

File: config/initializers/new_framework_defaults_7_2.rb [hunk 6]
```diff
@@ -0,0 +1,70 @@
+# Be sure to restart your server when you modify this file.
```
Comment (justinlittman) [inline config/initializers/new_framework_defaults_7_2.rb:1]: You can get rid of this file.

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comments.

File: config/environments/test.rb [hunk 7]
```diff
@@ -8,41 +8,45 @@
...
   # Show full error reports and disable caching.
-  config.consider_all_requests_local       = true
+  config.consider_all_requests_local = true
   config.action_controller.perform_caching = false
   config.cache_store = :null_store
 
-  # Raise exceptions instead of rendering exception templates.
-  config.action_dispatch.show_exceptions = false
+  # Render exception templates for rescuable exceptions and raise for other exceptions.
+  config.action_dispatch.show_exceptions = :rescuable
```
Comment (mjgiarlo) [inline config/environments/test.rb:29]: `false` is now `:none`, not `:rescuable`

File: public/robots.txt [hunk 8]
```diff
@@ -1,5 +1 @@
-# See http://www.robotstxt.org/wc/norobots.html for documentation on how to use the robots.txt file
-#
-# To ban all spiders from the entire site uncomment the next two lines:
-User-Agent: *
-Disallow: /
```
Comment (mjgiarlo) [inline public/robots.txt:5]: These lines were uncommented which makes me think we might still want them.

File: bin/setup (hunk 3 above)
Comment (peetucket) [reply bin/setup:5]: yeah, that is the historic name of the application, as shown in `config/application.rb` ... this was auto added by the script, presumably from that module name

File: config/application.rb (hunk 4 above)
Comment (peetucket) [reply config/application.rb:25]: thanks ... no, i'm not sure -- so i will put it back just in case

File: config/environments/production.rb (hunk 5 above)
Comment (peetucket) [reply config/environments/production.rb:25]: tests seem fine and i deployed to dev and that was fine too with some actual API calls I tried, so may not be needed (and sul-pub basically has no public web interface anyway)

File: config/environments/test.rb (hunk 7 above)
Comment (peetucket) [reply config/environments/test.rb:29]: ok, will change

**Pull request merged**